lib/editor/components/timetable/TimetableEditor.js

```diff
@@ -182,22 +183,46 @@ export default class TimetableEditor extends Component<Props, State> {
           const blankStopTime: StopTime = camelCaseKeys(generateNullProps('stoptime'))
           objectPath.set(newRow, `stopTimes.${i}`, blankStopTime)
         }
-        objectPath.set(newRow, `stopTimes.${i}.stopId`, stop.stopId)
-        cumulativeTravelTime += +stop.defaultTravelTime
+
+        // $FlowFixMe Flow doesn't recognize our check for stopId property?
+        objectPath.set(newRow, `stopTimes.${i}.stopId`, halt.hasOwnProperty('stopId') ? halt.stopId : halt.locationId)
```

Comment:
`halt.hasOwnProperty('stopId') ? halt.stopId : halt.locationId` => may be worth extracting a util method. I think it is being used multiple times.